test/e2e/syncer/syncer_test.go

```diff
@@ -208,4 +209,134 @@ func TestSyncWorkload(t *testing.T) {
+
+	ctx, cancelFunc := context.WithCancel(context.Background())
+	t.Cleanup(cancelFunc)
+
+	t.Log("Check initial workload")
+	cluster, err := kcpClient.WorkloadV1alpha1().WorkloadClusters().Get(ctx, workloadClusterName, metav1.GetOptions{})
+	require.NoError(t, err, "failed to get workload cluster", workloadClusterName)
+	require.Equal(t, false, cluster.Spec.Unschedulable)
+	require.Nil(t, cluster.Spec.EvictAfter)
+
+	t.Log("Cordon workload")
+	subCommandCordon := []string{
+		"workload",
+		"cordon",
+		workloadClusterName,
+	}
+	expected := []byte(workloadClusterName + " cordoned\n")
+	output := framework.RunKcpCliPlugin(t, kubeconfigPath, subCommandCordon)
+	require.Equal(t, output, expected)
+
+	t.Log("Check workload after cordon")
+	cluster, err = kcpClient.WorkloadV1alpha1().WorkloadClusters().Get(ctx, workloadClusterName, metav1.GetOptions{})
+	require.NoError(t, err, "failed to get workload cluster", workloadClusterName)
+	require.Equal(t, true, cluster.Spec.Unschedulable)
+	require.Nil(t, cluster.Spec.EvictAfter)
+
+	expected = []byte(workloadClusterName + " already cordoned\n")
+	output = framework.RunKcpCliPlugin(t, kubeconfigPath, subCommandCordon)
+	require.Equal(t, output, expected)
+
+	t.Log("Uncordon workload")
+	subCommandUncordon := []string{
+		"workload",
+		"uncordon",
+		workloadClusterName,
+	}
+	expected = []byte(workloadClusterName + " uncordoned\n")
+	output = framework.RunKcpCliPlugin(t, kubeconfigPath, subCommandUncordon)
+	require.Equal(t, output, expected)
+
+	t.Log("Check workload after uncordon")
+	cluster, err = kcpClient.WorkloadV1alpha1().WorkloadClusters().Get(ctx, workloadClusterName, metav1.GetOptions{})
+	require.NoError(t, err, "failed to get workload cluster", workloadClusterName)
+	require.Equal(t, false, cluster.Spec.Unschedulable)
+	require.Nil(t, cluster.Spec.EvictAfter)
+
+	expected = []byte(workloadClusterName + " already uncordoned\n")
+	output = framework.RunKcpCliPlugin(t, kubeconfigPath, subCommandUncordon)
+	require.Equal(t, output, expected)
+
+	t.Log("Drain workload")
+	subCommandDrain := []string{
+		"workload",
+		"drain",
+		workloadClusterName,
+	}
+	expected = []byte(workloadClusterName + " draining\n")
+	output = framework.RunKcpCliPlugin(t, kubeconfigPath, subCommandDrain)
+	require.Equal(t, output, expected)
+
+	t.Log("Check workload after drain started")
+	cluster, err = kcpClient.WorkloadV1alpha1().WorkloadClusters().Get(ctx, workloadClusterName, metav1.GetOptions{})
+	require.NoError(t, err, "failed to get workload cluster", workloadClusterName)
+	require.Equal(t, true, cluster.Spec.Unschedulable)
+	require.NotNil(t, cluster.Spec.EvictAfter)
+
+	expected = []byte(workloadClusterName + " already draining\n")
+	output = framework.RunKcpCliPlugin(t, kubeconfigPath, subCommandDrain)
+	require.Equal(t, output, expected)
+
+	//
+	// TODO: add a deployment before the drain and then check to see that the pods drained
+	//
```

Comment:
no, that's not this test's duty. Whether the API works is tested in API related tests.

Reply:
removed